src/components/ToolbarMenu.js

```diff
@@ -25,6 +25,10 @@ export class ToolbarMenu extends React.Component {
     this.state = { visible: false };
   }
 
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClick.bind(this));
```

Comment:
I am not sure that any of these `removeEventListener` calls will do what you want, because when you call `bind` you get a new function that is not `===`, which means that you won't remove anything here. (I could be wrong about that, but that's what I think will happen.)

I think the way to do this is as follows:

```
_subscribe() {
  if (!this._listener) {
    this._listener = this.handleClick.bind(this);
    document.addEventListener('click', this._listener);
  }
}

_release() {
  if (this._listener) {
    document.removeEventListener('click', this._listener);
    this._listener = null;
  }
}
```

And then call those methods.